extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/SmallRyeGraphQLClientRecorder.java

```diff
@@ -139,10 +142,32 @@ private GraphQLClientConfiguration toSmallRyeNativeConfiguration(GraphQLClientCo
                     transformed.setUsesSni(Boolean.valueOf(tlsConfiguration.usesSni()));
                 });
 
-        quarkusConfig.proxyHost().ifPresent(transformed::setProxyHost);
-        quarkusConfig.proxyPort().ifPresent(transformed::setProxyPort);
-        quarkusConfig.proxyUsername().ifPresent(transformed::setProxyUsername);
-        quarkusConfig.proxyPassword().ifPresent(transformed::setProxyPassword);
+        if (quarkusConfig.proxyHost().isPresent()) {
+            // use the deprecated proxy settings if they are set
+            quarkusConfig.proxyHost().ifPresent(transformed::setProxyHost);
+            quarkusConfig.proxyPort().ifPresent(transformed::setProxyPort);
+            quarkusConfig.proxyUsername().ifPresent(transformed::setProxyUsername);
+            quarkusConfig.proxyPassword().ifPresent(transformed::setProxyPassword);
+        } else {
+            // use the proxy configuration registry
+            resolveProxyConfiguration(quarkusConfig)
+                    .ifPresent(proxyConfiguration -> {
+                        transformed.setProxyHost(proxyConfiguration.host());
+                        transformed.setProxyPort(proxyConfiguration.port());
+                        proxyConfiguration.username().ifPresent(transformed::setProxyUsername);
+                        proxyConfiguration.password().ifPresent(transformed::setProxyPassword);
+                        // TODO: these properties will need to be supported on the smallrye-graphql side first
+                        proxyConfiguration.nonProxyHosts().ifPresent(x -> {
+                            org.jboss.logging.Logger.getLogger(this.getClass()).warn(
+                                    "Non-proxy hosts setting is not yet supported by the GraphQL client extension and will be ignored.");
+                        });
+                        if (proxyConfiguration.type() != ProxyType.HTTP) {
+                            throw new ConfigurationException(
+                                    "Only HTTP proxy type is supported by the GraphQL client extension at the moment.");
```

Comment:
Would it be clear from the stack trace or elsewhere, which specific config key is incompatible with this extension? We could perhaps also suggest some corrective action. How about

```
"Quarkus SmallRye GraphQL extension currently supports only proxy type 'HTTP'. You may either change the value of 'quarkus.proxy"+ (quarkusConfig.proxyConfigurationName().isEmpty() ? "" : ("." + quarkusConfig.proxyConfigurationName().get)) +".type' to 'HTTP', or you can create a new proxy configuration having 'quarkus.proxy.\"some-name\".type = HTTP' and set 'quarkus.graphql.proxy-configuration-name = some-name'.
```